src/macros.rs

```diff
@@ -1,199 +1,151 @@
-/// WARN:main: this is a warning message
-/// ```
+/// The standard logging macro.
 ///
-/// ```{.bash}
-/// $ RUST_LOG=debug ./main
-/// DEBUG:main: this is a debug message
-/// WARN:main: this is a warning message
-/// ```
-///
-/// ```{.bash}
-/// $ RUST_LOG=6 ./main
-/// DEBUG:main: this is a debug message
-/// WARN:main: this is a warning message
-/// 6:main: this is a custom logging level: 6
-/// ```
+/// This macro will generically log with the specified `LogLevel` and `format!`
+/// based argument list.
 #[macro_export]
 macro_rules! log {
     ($lvl:expr, $($arg:tt)+) => ({
-        static LOC: ::log::LogLocation = ::log::LogLocation {
+        static LOC: $crate::LogLocation = $crate::LogLocation {
             line: line!(),
             file: file!(),
             module_path: module_path!(),
         };
         let lvl = $lvl;
-        if log_enabled!(lvl) {
-            ::log::log(lvl, &LOC, format_args!($($arg)+));
+        if lvl <= $crate::max_log_level() {
+            $crate::log(lvl, &LOC, format_args!($($arg)+))
         }
     })
 }
 
-/// A convenience macro for logging at the error log level.
-///
-/// # Example
-///
-/// ```
-/// #[macro_use] extern crate log;
-///
-/// fn main() {
-///     let error = 3u;
-///     error!("the build has failed with error code: {}", error);
-/// }
-/// ```
-///
-/// Assumes the binary is `main`:
-///
-/// ```{.bash}
-/// $ RUST_LOG=error ./main
-/// ERROR:main: the build has failed with error code: 3
-/// ```
+/// Logs a message at the error level.
 ///
+/// Logging through this macro is disabled if the `log_level = "off"` cfg is
+/// present.
 #[macro_export]
 macro_rules! error {
-    ($($arg:tt)*) => (log!(::log::ERROR, $($arg)*))
+    ($($arg:tt)*) => (
+        match () {
+            #[cfg(not(any(log_level = "off")))]
+            () => log!($crate::LogLevel::Error, $($arg)*),
+            #[cfg(any(log_level = "off"))]
+            () => {}
+        }
+    )
 }
 
-/// A convenience macro for logging at the warning log level.
-///
-/// # Example
-///
-/// ```
-/// #[macro_use] extern crate log;
-///
-/// fn main() {
-///     let code = 3u;
-///     warn!("you may like to know that a process exited with: {}", code);
-/// }
-/// ```
-///
-/// Assumes the binary is `main`:
+/// Logs a message at the warn level.
 ///
-/// ```{.bash}
-/// $ RUST_LOG=warn ./main
-/// WARN:main: you may like to know that a process exited with: 3
-/// ```
+/// Logging through this macro is disabled if any of the following cfgs are
+/// present: `log_level = "off"` or `log_level = "error"`.
 #[macro_export]
 macro_rules! warn {
-    ($($arg:tt)*) => (log!(::log::WARN, $($arg)*))
+    ($($arg:tt)*) => (
+        match () {
+            #[cfg(not(any(log_level = "off",
+                          log_level = "error")))]
+            () => log!($crate::LogLevel::Warn, $($arg)*),
+            #[cfg(any(log_level = "off",
+                      log_level = "error"))]
+            () => {}
+        }
```

Comment:
I ended up switching from `if cfg!()` to this to ensure that the logging statement is entirely removed early in the compilation process if it's statically disabled, but the unused variable thing is a good point. It might not be worth the trouble.